src/btree/bt_sync.c

```diff
@@ -382,6 +419,18 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
                 if (conn->ckpt_write_pages % 5000 == 0)
                     __wt_checkpoint_progress(session, false);
             }
+
+            /* Reconciliation of a history store page resulted as empty, mark it for deletion. */
+            if (is_las && page->modify != NULL && page->modify->rec_result == WT_PM_REC_EMPTY) {
+                /*
+                 * The duplicate tree walk code expects the ref state to be in memory. We need to
+                 * get the duplicate tree walk pointer before marking the current tree page as
+                 * deleted.
+                 */
+                WT_ERR(__sync_dup_walk(session, walk, flags, &prev));
+                WT_ERR(__sync_ref_mark_deleted(session, walk));
+                goto skipwalk;
+            }
```

Comment:
If we reconcile a history page and there are no records to write, why does that mean we should mark it for deletion? Imagine a page with 5 records, none of which need to be written, so we get a `WT_PM_REC_EMPTY` result, but after the reconciliation completes, additional records are inserted. Is there something about the insert pattern of the history file that makes that impossible?

Reply:
I understood the race condition that is possible in marking deleted of an in-memory page when it is used for insertion. So marking to be deleted after reconciliation is removed. Take care of these in-memory pages when we traversing the parent page.